lib/smart_answer/money.rb

```diff
@@ -32,8 +25,21 @@ def <=>(other)
       end
     end
 
-    def self.valid?(raw_input)
-      raw_input.is_a?(Numeric) || raw_input.is_a?(Money) || raw_input =~ /\A *[0-9]+(\.[0-9]{1,2})? *\z/
+    def self.raw_input_to_number(raw_input)
+      raw_input = raw_input.is_a?(Numeric) ? raw_input : raw_input.to_s.delete(',').gsub(/\s/, '')
+      if raw_input.to_f.infinite?
+        raise InvalidResponse, "Sorry, that number is too big. Please try again.", caller
+      else
+        valid_raw_input(raw_input)
+      end
+    end
+
+    def self.valid_raw_input(raw_input)
+      if raw_input.is_a?(Numeric) || raw_input.is_a?(Money) || raw_input =~ /\A *[0-9]+(\.[0-9]{1,2})? *\z/
+        raw_input
+      else
+        raise InvalidResponse, "Sorry, I couldn't understand that number. Please try again.", caller
```

Comment:
(I know this isn't  part of your change, but while we're here, do you think "**I** couldn't understand that number" is a normal sort of error message to return? I don't think I'd expect a validation message to use the first person. Maybe more like "The entered value is not a valid number. Please try again."?)